code/game/machinery/newscaster/newscaster_machine.dm

```diff
@@ -0,0 +1,619 @@
+	else
+		data["user"] = list()
+		data["user"]["name"] = user.name
+		data["user"]["job"] = "N/A"
+		data["user"]["department"] = "N/A"
+
+	data["security_mode"] = (ACCESS_ARMORY in card?.GetAccess())
+	data["photo_data"] = !isnull(current_image)
+	data["creating_channel"] = creating_channel
+	data["creating_comment"] = creating_comment
+
+	//Code breaking down the channels that have been made on-station thus far. ha
+	//Then, breaks down the messages that have been made on those channels.
+	for(var/datum/newscaster/feed_channel/channel as anything in GLOB.news_network.network_channels)
+		channel_list += list(list(
+			"name" = channel.channel_name,
+			"author" = channel.author,
+			"censored" = channel.censored,
+			"locked" = channel.locked,
+			"ID" = channel.channel_ID,
+		))
+	if(current_channel)
+		for(var/datum/newscaster/feed_message/feed_message as anything in current_channel.messages)
+			var/photo_ID = null
+			var/list/comment_list
+			if(feed_message.img)
+				user << browse_rsc(feed_message.img, "tmp_photo[feed_message.message_ID].png")
+				photo_ID = "tmp_photo[feed_message.message_ID].png"
+			for(var/datum/newscaster/feed_comment/comment_message as anything in feed_message.comments)
+				comment_list += list(list(
+					"auth" = comment_message.author,
+					"body" = comment_message.body,
+					"time" = comment_message.time_stamp,
+				))
+			message_list += list(list(
+				"auth" = feed_message.author,
+				"body" = feed_message.body,
+				"time" = feed_message.time_stamp,
+				"channel_num" = feed_message.parent_ID,
+				"censored_message" = feed_message.bodyCensor,
+				"censored_author" = feed_message.authorCensor,
+				"ID" = feed_message.message_ID,
+				"photo" = photo_ID,
+				"comments" = comment_list
+			))
+
+
+	data["viewing_channel"] = current_channel?.channel_ID
+	data["paper"] = paper_remaining
+	//Here we display all the information about the current channel.
+	data["channelName"] = current_channel?.channel_name
+	data["channelAuthor"] = current_channel?.author
+
+	if(!current_channel)
+		data["channelAuthor"] = "Nanotrasen Inc"
+		data["channelDesc"] = "Welcome to Newscaster Net. Interface & News networks Operational."
+		data["channelBlocked"] = TRUE
+	else
+		data["channelDesc"] = current_channel.channel_desc
+		data["channelBlocked"] = (current_channel.locked || current_channel.censored)
+		data["channelCensored"] = current_channel.censored
+
+	//We send all the information about all channels and all messages in existance.
+	data["channels"] = channel_list
+	data["messages"] = message_list
+
+	var/list/formatted_requests = list()
+	var/list/formatted_applicants = list()
+	for (var/datum/station_request/request as anything in GLOB.request_list)
+		formatted_requests += list(list("owner" = request.owner, "value" = request.value, "description" = request.description, "acc_number" = request.req_number))
+		if(request.applicants)
+			for(var/datum/bank_account/bank_account_b as anything in request.applicants)
```

Comment:
What's this name meant to mean?

Reply:
Couldn't think of a better name for an iterator variable for bank account so `bank_account_b` seemed unique enough. What would you prefer?

Reply:
applicant_bank_account too much for you?